### PR TITLE
Single video page

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features: 
- on clicking video thumbnail, it redirects to single video page

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import { Routes, Route } from "react-router-dom";
 
 function App() {
   return (
     <div className="App">
-      <VideoListingPage/>
+      <Routes>
+        <Route path="/" element={<Homepage />} />
+        <Route path="/explore" element={<VideoListingPage />} />
+        <Route path="/:videoId" element={<SingleVideoPage />} />
```
Comment (MarkVed17) [inline src/App.jsx:11]: Shouldn’t the path be ‘/explore/:videoId’ ?

File: src/Pages/Homepage.jsx [hunk 2]
```diff
@@ -1,71 +1,77 @@
 import axios from "axios";
 import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
 import { Category, Navbar, VideoThumbnail } from "../Components";
-import "./CSS/homepage.css"
+import "./CSS/homepage.css";
 
 export function Homepage() {
+  const [categories, setCategories] = useState([]);
+  const [videos, setVideos] = useState([]);
```
Comment (MarkVed17) [inline src/Pages/Homepage.jsx:9]: You can maintain ‘state’ globally for the videos, categories and such data.

**Review by MarkVed17 — APPROVED** [review]

File: src/Pages/Homepage.jsx (hunk 2 above)
Comment (t007rushi) [reply src/Pages/Homepage.jsx:9]: rightly pointed out by @MarkVed17

File: src/Pages/Homepage.jsx [hunk 3]
```diff
@@ -1,71 +1,77 @@
...
-        try {
-            const response = await axios.get("/api/categories");
-            setCategories(response.data.categories);
-        } 
-        catch (error) {
-            console.log(error);
-        }
+  const getCategories = async () => {
+    try {
+      const response = await axios.get("/api/categories");
```
Comment (t007rushi) [inline src/Pages/Homepage.jsx:13]: destructure here itself ```{data}```

**Review by t007rushi — APPROVED** [review]
Looks good to merge here

**Review by manojasarsa — APPROVED** [review]
The single video page is working as expected. 
The code seems clean and structured.
GTG!

**Pull request merged**